step_clone_vm.go

```diff
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"github.com/vmware/govmomi"
+	"context"
+	"github.com/mitchellh/multistep"
+	"github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/govmomi/object"
+	"github.com/hashicorp/packer/packer"
+	"strconv"
+)
+
+type CloningEnv struct {
+	client *govmomi.Client
+	folder *object.Folder
+	vm_src *object.VirtualMachine
+	ctx    context.Context
+}
+
+func NewCloningEnv(state multistep.StateBag) *CloningEnv {
+	env := new(CloningEnv)
+	env.client = state.Get("client").(*govmomi.Client)
+	env.folder = state.Get("folder").(*object.Folder)
+	env.vm_src = state.Get("vm_src").(*object.VirtualMachine)
+	env.ctx = state.Get("ctx").(context.Context)
+	return env
+}
+
+type StepCloneVM struct{
+	config *Config
+	//vm_custom VMCustomParams
+	success bool
+}
+
+func (s *StepCloneVM) Run(state multistep.StateBag) multistep.StepAction {
+	ui := state.Get("ui").(packer.Ui)
+	ui.Say("start cloning...")
+
+	env := NewCloningEnv(state)
+	vm, err := CloneVM(s.config, env)
```

Comment:
can we declare a single parameter structure, where all values are listed explicitly